Cisto malo da ulepsam kod

Igracx opened the pull request:
Prebacio sam linije sa pocetka u header.py i onda sam svuda importovao sa from header import * takodje sam dodao os da pygame ne bi prikazivao poruku i preimenovao sam main.py u main.pyw da se ne bi otvarala konzola i stavio sam caption na Dinochrome umesto pygame, ovo je vise test da se snadjem sa gitom. E da kada je dinosaurus u skoku i kada pritsnes strelicu na dole spusta se brze nego sto treba, iako ovako nije u originalnoj igrici bolje neka ostane da bude zanimljivije. Pozz

Igracx committed:
Prebacio sam linije sa pocetka u header.py i onda sam svuda importovao sa from header import * takodje sam dodao os da pygame ne bi prikazivao poruku i preimenovao sam main.py u main.pyw da se ne bi otvarala konzola, ovo je vise test da se snadjem sa gitom. E da kada je dinosaurus u skoku i kada pritsnes strelicu na dole vrati se brze nego sto treba, iako ovako nije u originalnoj igrici bolje neka ostane da bude zanimljivije. Pozz

Vasiljko reviewed (changes requested):
Change line 11 in main.pyw to `PTERO_POS = [180,150,100]`

Igracx committed:
node_genes nije recnik oblika {broj:"tip neurona"} jer je redosled u
neatu bitan umesto toga je u obliku numpy array-a gde je svaki red par
(broj, "tip neurona"), prvo idu sensor neuroni pa output neuroni, i
nakon njih ce se ubacivati hidden neuroni, connection_genes je takodje
numpy array gde je svaki red petorka oblika (in-node, out-node,
weight-value, enable-bit, innovation number), jos sam napravio i
izmenjenu sigmoidalnu funkciju kako su je trazili u papiru i to je to
sto sam ja radio za ova dva dana hahahaha :'( ali dobro sta da se radi
nasao sam na
https://stackoverflow.com/questions/55569260/feedforward-algorithm-in-neat-neural-evolution-of-augmenting-topologies
objasnjeno kako treba da izgleda taj feedforward. Pozz.

Igracx committed:
Added starting initialization for Agent class
as well as copy genes from a parent.
Neat starts out from minimal structure with each sensor node connected
to each output node, in the connection_genes first node one is connected
to each output node in order in which they apper (so first to node with
the lowest identification number (example 4, than to 5 then to 6 and so
on) and node 2 is connected to each output node and so on.
Each connection gets a random weight from uniform distribution from
interval [-5, 5), I don't know how optimal this range is but none was
given in the paper.
If class is given one parent it will copy genes from that parent, and if it is given
two parents than it will be made as a result of cross over which remains
to be implemented.
